File: 23) Is subsequence/isSubseq.py
def is_subsequence(s, t):
    i, j = 0, 0
    while i < len(s) and j < len(t):
        if s[i] == t[j]:
            i += 1
        j += 1
    return i == len(s)
s = input("Enter s: ")
t = input("Enter t: ")
print(is_subsequence(s, t)) 

File: 23) Is subsequence/isSubseq.cpp
#include <iostream>
using namespace std;

bool isSubsequence(string s, string t) {
    int i = 0, j = 0;
    while (i < s.length() && j < t.length()) {
        if (s[i] == t[j]) {
            i++;
        }
        j++;
    }
    return i == s.length();
}

int main() {
    string s, t;
    cout << "Enter s: ";
    getline(cin, s);
    cout << "Enter t: ";
    getline(cin, t);
    cout << isSubsequence(s, t) << endl;
    return 0;
}
